File: dataset/human/java/file_323.java
import java.util.*;
import java.util.ArrayList;

class Main{
    public static void main(String[] args){
        
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int[] A = new int[N];
        int m = 0;
        int min = 1000000001;
        long sum = 0;

        for(int i=0; i<N; i++){
            A[i] = sc.nextInt();
            sum += Math.abs(A[i]);
            min = Math.min(min, Math.abs(A[i]));
            if(A[i]<0){
                m += 1;
            }
        }
        if(m%2==0){
            System.out.println(sum);
        }
        else{
            System.out.println(sum - 2 * min);
        }
    }
}

File: dataset/human/python/file_323.py
from sys import stdin
N = int(stdin.readline().rstrip())
A = [int(_) for _ in stdin.readline().rstrip().split()]
minus = []
for i in range(N):
  if A[i] < 0:
    minus.append(A[i])
abs_A = [abs(i) for i in A]
if len(minus) % 2 == 0:
  print(sum(abs_A))
else:
  print(sum(abs_A)-min(abs_A)*2)
